metrics/cli/test/gcloud/adapter/gcloud_cli_service_adapter_test.dart

```diff
@@ -0,0 +1,157 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:cli/gcloud/adapter/gcloud_cli_service_adapter.dart';
+import 'package:cli/gcloud/cli/gcloud_cli.dart';
+import 'package:cli/prompt/prompter.dart';
+import 'package:cli/prompt/strings/gcloud_strings.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../test_utils/matchers.dart';
+import '../../test_utils/prompt_writer_mock.dart';
+
+void main() {
+  group('GCloudCliServiceAdapter', () {
+    const region = 'testRegion';
+
+    final gcloudCli = _GCloudCliMock();
+    final promptWriter = PromptWriterMock();
+    final prompter = Prompter(promptWriter);
+    final gcloudAdapter = GCloudCliServiceAdapter(gcloudCli, prompter);
+
+    tearDown(() {
+      reset(gcloudCli);
+      reset(promptWriter);
+    });
+
+    test(
+      "throws an ArgumentError if the given gcloud CLI is null",
+      () {
+        expect(
+          () => GCloudCliServiceAdapter(null, prompter),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given prompter is null",
+      () {
+        expect(
+          () => GCloudCliServiceAdapter(gcloudCli, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".login() logins to the GCloud CLI",
+      () async {
+        await gcloudAdapter.login();
+
+        verify(gcloudCli.login()).called(once);
+      },
+    );
+
+    test(
+      ".createProject() shows available regions before requesting the region from the user",
```

Comment:
I think it would be great to test that each method called only once. What do you think?